models/action_models.go

```diff
@@ -0,0 +1,147 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+//go:generate reform
+
+// ActionResult describes an action result which is storing in persistent storage.
+//reform:action_results
+type ActionResult struct {
+	ID         string    `reform:"id,pk"`
+	PmmAgentID string    `reform:"pmm_agent_id"`
+	Done       bool      `reform:"done"`
+	Error      string    `reform:"error"`
+	Output     string    `reform:"output"`
+	CreatedAt  time.Time `reform:"created_at"`
+	UpdatedAt  time.Time `reform:"updated_at"`
+}
+
+// BeforeInsert implements reform.BeforeInserter interface.
+//nolint:unparam
+func (s *ActionResult) BeforeInsert() error {
+	now := Now()
+	s.CreatedAt = now
+	s.UpdatedAt = now
+	return nil
+}
+
+// BeforeUpdate implements reform.BeforeUpdater interface.
+//nolint:unparam
+func (s *ActionResult) BeforeUpdate() error {
+	s.UpdatedAt = Now()
+	return nil
+}
+
+// AfterFind implements reform.AfterFinder interface.
+//nolint:unparam
+func (s *ActionResult) AfterFind() error {
+	s.CreatedAt = s.CreatedAt.UTC()
+	s.UpdatedAt = s.UpdatedAt.UTC()
+	return nil
+}
+
+// PtSummaryAction represents pt-summary domain model.
```

Comment:
Sorry, but `models` is not a package for the domain model. It is really just a small data access layer.
It seems that `PtSummaryAction` and other types below belong to the service package. `Args()` definitely does not belong there.

Reply:
@AlekSi I've moved them to service...
```
main_test.go:104: Package "github.com/percona/pmm-managed/services/agents" should not import package "github.com/percona/pmm-managed/services/action" (blacklisted by "github.com/percona/pmm-managed/services").
```

https://travis-ci.com/percona/pmm-managed/jobs/203514513